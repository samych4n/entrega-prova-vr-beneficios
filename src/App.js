import React from 'react';
import './styles/App.css';
import Pokemon from './pokemon-react';
import { Provider } from 'react-redux';
import store from './store';

function App() {
	return (
		<div className="App">
			<header className="App-header" />
			<Provider store={store}>
				<Pokemon />
			</Provider>
		</div>
	);
}

export default App;
