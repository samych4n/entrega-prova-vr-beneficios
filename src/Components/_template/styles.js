import styled from 'styled-components';

export const CardSquare = styled.div``;
