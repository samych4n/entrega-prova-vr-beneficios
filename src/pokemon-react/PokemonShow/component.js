import React from 'react';
import {
	StatsName,
	StatContainer,
	StatsBar,
	StatsValue,
	TypesContainer,
	Types,
	AbilitiesContainer,
	AbilitiesTitle,
	Abilities,
	GamesContainer,
	GamesNameContainer,
	GamesTitle,
	Games,
	SpritesContainer,
	SpritesGroupContainer,
	SpritesSexContainer,
	SpritesTitle,
	Sprites,
	MovesContainer,
	MovesNameContainer,
	MovesTitle,
	Moves,
} from './styles';

import { statColor, typeColor } from '../../configs/colors';
export function Stat({ name, value }) {
	return (
		<StatContainer>
			<StatsName>{name}</StatsName>
			<StatsValue>{value}</StatsValue>
			<StatsBar value={value} color={statColor[name] || '#FFFFFF'} />
		</StatContainer>
	);
}

export function Type({ types }) {
	return (
		<TypesContainer>
			{types
				.sort((a, b) => (a.slot < b.slot ? 1 : -1))
				.map(val => (
					<Types
						key={val.type.name}
						color={typeColor[val.type.name] || '#FFFFFF'}
					>
						{val.type.name}
					</Types>
				))}
		</TypesContainer>
	);
}

export function Ability({ abilities }) {
	return (
		<AbilitiesContainer>
			<AbilitiesTitle>Habilidades</AbilitiesTitle>
			{abilities.map(ability => (
				<Abilities
					key={ability.ability.name}
					is_hidden={ability.is_hidden}
				>
					{ability.ability.name}
				</Abilities>
			))}
		</AbilitiesContainer>
	);
}

export function Game({ games }) {
	const gamesInfo = games
		.map(game => ({
			...game,
			id_game: /https:\/\/pokeapi.co\/api\/v2\/version\/([0-9]*)\//.exec(
				game.version.url
			)[1],
		}))
		.sort((a, b) => a.id_game - b.id_game);
	return (
		<GamesContainer>
			<GamesTitle>Jogos</GamesTitle>
			<GamesNameContainer>
				{gamesInfo.map(game => (
					<Games key={game.id_game}>{game.version.name}</Games>
				))}
			</GamesNameContainer>
		</GamesContainer>
	);
}

export function Sprite({ sprites }) {
	return (
		<SpritesContainer>
			<SpritesTitle>Sprites</SpritesTitle>
			<SpritesSexContainer>
				<SpritesGroupContainer>
					<Sprites src={sprites.front_default} />
					<Sprites src={sprites.back_default} />
				</SpritesGroupContainer>
				<SpritesGroupContainer>
					<Sprites src={sprites.front_shiny} />
					<Sprites src={sprites.back_shiny} />
				</SpritesGroupContainer>
			</SpritesSexContainer>
			{sprites.front_female && (
				<SpritesSexContainer>
					<SpritesGroupContainer>
						<Sprites src={sprites.front_female} />
						<Sprites src={sprites.back_female} />
					</SpritesGroupContainer>
					<SpritesGroupContainer>
						<Sprites src={sprites.front_shiny_female} />
						<Sprites src={sprites.back_shiny_female} />
					</SpritesGroupContainer>
				</SpritesSexContainer>
			)}
		</SpritesContainer>
	);
}

export function Move({ moves }) {
	return (
		<MovesContainer>
			<MovesTitle>Moves</MovesTitle>
			<MovesNameContainer>
				{moves.map(move => (
					<Moves key={move.move.name}>{move.move.name}</Moves>
				))}
			</MovesNameContainer>
		</MovesContainer>
	);
}
