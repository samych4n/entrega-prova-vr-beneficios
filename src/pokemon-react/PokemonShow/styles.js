import styled from 'styled-components';

export const PokemonShowContainer = styled.div`
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	width: 100%;
`;

export const Section = styled.div`
	display: flex;
	border-radius: 15px;
	border: 8px solid #df1818;
	background-color: white;
	margin-bottom: 20px;
	padding: 10px;
`;

export const Nome = styled(Section)`
	width: 70%;
	max-width: 800px;

	font-size: 20px;
	font-weight: bold;
	text-transform: uppercase;
	justify-content: center;

	@media (max-width: 500px) {
		width: 85%;
	}
`;

export const Picture = styled(Section)`
	width: 70%;
	&:after {
		content: '';
		display: block;
		padding-bottom: 100%;
	}
	& img {
		height: 100%;
		width: 100%;
	}
	@media (max-width: 500px) {
		width: 85%;
	}
`;

// #region stats
export const Stats = styled(Section)`
	display: flex;
	flex-direction: column-reverse;
	width: 70%;
	@media (max-width: 500px) {
		width: 85%;
	}
`;
export const StatContainer = styled.div`
	display: flex;
	justify-content: space-between;
	position: relative;
	margin: 2.5px;
	border: 1px solid black;
	border-radius: 15px;
`;
export const StatsName = styled.div`
	margin-left: 10px;
	font-weight: bold;
	font-size: 15px;
`;

export const StatsValue = styled.div`
	font-size: 15px;
	font-weight: bold;
	margin-right: 10px;
`;

export const StatsBar = styled.div`
	font-size: 10px;
	background-color: ${props => props.color};
	position: absolute;
	height: 100%;
	opacity: 0.5;
	border-radius: 15px;
	width: ${props => `calc(100% * ${props.value} / 255)`};
`;

//#endregion stats

// #region types
export const TypesContainer = styled(Section)`
	display: flex;
	flex-direction: row;
	justify-content: space-evenly;
	width: 70%;
	@media (max-width: 500px) {
		width: 85%;
	}
`;

export const Types = styled.div`
	background-color: ${props => props.color};
	font-size: 20px;
	font-weight: bold;
	padding: 5px;
	flex: 1;
	text-align: center;
`;

// #endregion types

// #region abilities
export const AbilitiesContainer = styled(Section)`
	width: 70%;
	display: flex;
	flex-direction: column;
	@media (max-width: 500px) {
		width: 85%;
	}
`;

export const AbilitiesTitle = styled.div`
	font-size: 20px;
	font-weight: bold;
	text-align: center;
	text-transform: uppercase;
	text-decoration: underline;
	padding-bottom: 5px;
`;

export const Abilities = styled.div`
	background-color: ${props => (props.is_hidden ? '#a67c00' : '#FFFFFF')};
	font-size: 15px;
	font-weight: bold;
	padding: 5px;
`;
// #endregion abilities

// #region games
export const GamesContainer = styled(Section)`
	width: 70%;
	display: flex;
	flex-direction: column;
	@media (max-width: 500px) {
		width: 85%;
	}
`;

export const GamesTitle = styled.div`
	font-size: 20px;
	font-weight: bold;
	text-align: center;
	text-transform: uppercase;
	text-decoration: underline;
	padding-bottom: 5px;
`;

export const GamesNameContainer = styled.div`
	font-size: 15px;
	font-weight: bold;
	padding: 5px;
	display: flex;
	flex-direction: row;
	flex-flow: wrap;
`;

export const Games = styled.div`
	font-size: 15px;
	font-weight: bold;
	padding: 5px;
`;
// #endregion games

// #region sprites
export const SpritesContainer = styled(Section)`
	width: 70%;
	display: flex;
	flex-direction: column;
	@media (max-width: 500px) {
		width: 85%;
	}
`;

export const SpritesTitle = styled.div`
	font-size: 20px;
	font-weight: bold;
	text-align: center;
	text-transform: uppercase;
	text-decoration: underline;
	padding-bottom: 5px;
`;

export const SpritesSexContainer = styled.div`
	padding: 5px;
	display: flex;
	flex-direction: row;
	flex-flow: wrap;
	justify-content: center;
`;

export const SpritesGroupContainer = styled.div`
	padding: 5px;
	display: flex;
	flex-flow: wrap;
	flex-direction: row;
	justify-content: center;
`;

export const Sprites = styled.img`
	&:after {
		content: '';
		display: block;
		padding-bottom: 100%;
	}
`;
// #endregion sprites
// #region moves
export const MovesContainer = styled(Section)`
	width: 70%;
	display: flex;
	flex-direction: column;
	@media (max-width: 500px) {
		width: 85%;
	}
`;

export const MovesTitle = styled.div`
	font-size: 20px;
	font-weight: bold;
	text-align: center;
	text-transform: uppercase;
	text-decoration: underline;
	padding-bottom: 5px;
`;

export const MovesNameContainer = styled.div`
	font-size: 15px;
	font-weight: bold;
	padding: 5px;
	display: flex;
	flex-direction: row;
	flex-flow: wrap;
`;

export const Moves = styled.div`
	font-size: 15px;
	font-weight: bold;
	padding: 5px;
`;
// #endregion moves
