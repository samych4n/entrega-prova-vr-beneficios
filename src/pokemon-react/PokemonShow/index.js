import React, { Component } from 'react';
import PokemonActions from '../../store/Actions/pokemon';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { getPokemonInfo } from '../../services/pokeapi';

import { Picture, PokemonShowContainer, Nome, Stats } from './styles';
import { Stat, Type, Ability, Game, Sprite, Move } from './component';

class PokemonShow extends Component {
	componentDidMount() {
		window.scrollTo(0, 0);
		this.loadPokemonInfo();
	}

	async loadPokemonInfo() {
		const { pokemonInfo, addPokemonInfo, match, history } = this.props;
		const { id } = match.params;
		if (!pokemonInfo[id]) {
			try {
				const info = await getPokemonInfo(id);
				addPokemonInfo(id, info);
			} catch (e) {
				history.push('/');
			}
		}
	}

	render() {
		const { match, pokemonInfo } = this.props;
		const { id } = match.params;
		if (!pokemonInfo[id])
			return (
				<PokemonShowContainer>Carregando Dados</PokemonShowContainer>
			);
		const pokemon = pokemonInfo[id];
		return (
			<PokemonShowContainer>
				<Nome>
					#{id}-{pokemon.species.name}
				</Nome>
				<Picture>
					<img
						src={pokemon.sprites.front_default || '/notfound.png'}
						alt={`imagem de ${pokemon.species.name}`}
					/>
				</Picture>
				<Type types={pokemon.types} />
				<Stats>
					{pokemon.stats.map(val => (
						<Stat
							key={val.stat.name}
							name={val.stat.name}
							value={val.base_stat}
						/>
					))}
				</Stats>
				<Ability abilities={pokemon.abilities} />
				<Sprite sprites={pokemon.sprites} />
				<Game games={pokemon.game_indices} />
				<Move moves={pokemon.moves} />
			</PokemonShowContainer>
		);
	}
}

const mapStateToProps = state => ({
	pokemonInfo: state.pokemon.pokemonInfo,
});

export default connect(
	mapStateToProps,
	PokemonActions
)(withRouter(PokemonShow));
