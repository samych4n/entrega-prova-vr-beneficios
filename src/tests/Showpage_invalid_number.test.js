/* eslint-disable no-undef */
import App from '../App';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import { PokemonShowContainer } from '../pokemon-react/PokemonShow/styles';

import { Stats } from '../pokemon-react/PokemonShow/styles';
import {
	Stat,
	Type,
	Ability,
	Game,
	Sprite,
	Move,
} from '../pokemon-react/PokemonShow/component';
import toJson from 'enzyme-to-json';

const noop = () => {};
Object.defineProperty(window, 'scrollTo', { value: noop, writable: true });
it('Showpage - Bulbassaur', async () => {
	const wrapper = mount(
		<MemoryRouter initialEntries={['/pokemon/2000']}>
			<App />
		</MemoryRouter>
	);
	await new Promise(res => setTimeout(res, 1000));
	wrapper.update();
	expect(toJson(wrapper.find(PokemonShowContainer))).toMatchSnapshot();

	expect(wrapper.find(Stats)).toHaveLength(0);
	expect(wrapper.find(Stat)).toHaveLength(0);
	expect(wrapper.find(Type)).toHaveLength(0);
	expect(wrapper.find(Ability)).toHaveLength(0);
	expect(wrapper.find(Game)).toHaveLength(0);
	expect(wrapper.find(Move)).toHaveLength(0);
	expect(wrapper.find(Sprite)).toHaveLength(0);
});
