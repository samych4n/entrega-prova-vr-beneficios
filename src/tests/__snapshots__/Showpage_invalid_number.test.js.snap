// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Showpage - Bulbassaur 1`] = `null`;
